packages/webview_flutter/lib/webview_flutter.dart

```diff
@@ -254,8 +254,22 @@ class WebViewController {
     return _channel.invokeMethod("reload");
   }
 
-  Future<void> _updateSettings(Map<String, dynamic> update) async {
-    return _channel.invokeMethod('updateSettings', update);
+  Future<void> _updateSettings(_WebSettings update) async {
+    Map<String, dynamic> updateMap;
+    if (_settings == null) {
+      updateMap = update.toMap();
+    } else {
+      updateMap = _settings.updatesMap(update);
+    }
+    _settings = update;
+    return _channel.invokeMethod('updateSettings', updateMap);
+  }
+
+  /// Evaluates JavaScript in the context of the current page.
+  /// Returns a Future containing the result of the JavaScript execution.
```

Comment:
Not sure what's a result of a JavaScript execution? would it be correct to say "... the result of evaluating the JavaScript expression"?